feature_engine/selection/drop_psi_features.py

```diff
@@ -464,6 +468,7 @@ def _observation_frequency_per_bin(self, basis, test):
                 how="outer",
             )
             .fillna(self.min_pct_empty_bins)
+            .replace(to_replace=0, value=self.min_pct_empty_bins)
```

Comment:
why do we add this line?

Reply:
If a categorical variable is of type 'object', and the split misses a category, the dataframe will have a NaN which is then solved by .fillna. However, in case the categorical variable is of type 'category', and the split misses a category, it will be correctly put to 0: thus it won't get filled and the PSI becomes infinite. 

This line ensures that empty buckets have a minimal value such that the PSI doesn't explode, similar to how it functions for 'object' type categoricals.